src/babashka/nrepl/impl/server.clj

```diff
@@ -162,14 +163,53 @@
                      "type" (cond
                               (ifn? (:val m)) "function"
                               :else "variable")
-                     "docstring" (:doc m)
+                     "docstring" (:doc m "")
                      "status" #{"done"}}]
           (utils/send os
                       (utils/response-for msg reply) opts)))
       (catch Throwable e
         (when debug (println e))
         (utils/send os (utils/response-for msg {"status" #{"done" "no-eldoc"}}) opts)))))
 
+(defn lookup [ctx os msg {:keys [debug] :as opts}]
+  (try
+    (let [ns-str (:ns msg)
+          sym-str (:symbol msg)
+          sym-ns (some-> (symbol sym-str)
+                         namespace
+                         symbol)
+          curr-ns-aliases (sci/eval-string*
```

Comment:
Why do you need to get the aliases first and can't you just use `resolve`?

E.g.:

```
user=> (resolve 'set/union)
#'clojure.set/union
```

Reply:
I tried rewriting the whole function using retrieval of the actual symbol using 

```clj
(let [[sci-sym-ns sci-sym-name] (sci/eval-string* ctx (format "(some->> (resolve '%s)
                                                                         symbol
                                                                         ((juxt namespace name)))" sym-str))])
```

This strategy seems to work and surprisingly seems to cover all the use-cases I was testing (both in IDE and in unit tests)

But..

1) I am actually not using `ns-str (:ns msg)` binding at all. Not sure if I have to. Perhaps it's not even needed as the `sym-str (:sym msg)` seems to come fully qualified anyways. But then again maybe that's only how calva sends it, IDK.

2) There also doesn't seem to be any obvious effect/benefit of following code fragments 

```clj
sci-ns (when ns-str
                 (sci-utils/namespace-object (:env ctx) (symbol ns-str) nil false))]
;; which is then used in 
 (sci/binding [vars/current-ns (or sci-ns @vars/current-ns)]
```

What's the purpose of `sci-ns` here, can't the whole `sci/binding` be ommited? I must be missing something.

Reply:
Basically this alternative implementation of `lookup` also works (in both IDE and tests), but I don't understand if `sci-utils/namespace-object` should be used in this scenario or not and if so, then how to use it correctly.

```clj
(defn lookup [ctx os msg {:keys [debug] :as opts}]
  (try
    (let [ns-str (:ns msg)
          sym-str (:symbol msg)
          [sci-sym-ns sci-sym-name] (sci/eval-string* ctx (format "(some->> (resolve '%s)
                                                                   symbol
                                                                   ((juxt namespace name)))" sym-str))
          sci-ns (when ns-str
                   (sci-utils/namespace-object (:env ctx) (symbol ns-str) nil false))
          m (sci/binding [vars/current-ns (or sci-ns @vars/current-ns)]
              (sci/eval-string*
               ctx
               (format "
(when-let [v (ns-resolve '%s '%s)]
  (let [m (meta v)]
    (assoc m :arglists (:arglists m)     
     :doc (:doc m)
     :name (:name m)
     :ns (some-> m :ns ns-name)
     )))"
                       sci-sym-ns
                       sci-sym-name)))
          reply {"ns" (:ns m)
                 "name" (:name m)
                 "arglists-str" (str (mapv #(mapv str %) (:arglists m)))
                 "doc" (:doc m "")
                 "status" #{"done"}}]
      (utils/send os
                  (utils/response-for msg reply) opts))
    (catch Throwable e
      (when debug (println e))
      (utils/send os (utils/response-for msg {"status" #{"done"}}) opts))))
```

Reply:
The ns binding is needed to evaluate code from within that ns. Although technically you don't need that when you use `ns-resolve`, the binding is also need in case the user switches the ns in some expression.

I suggest just copying the code from `eldoc`. But looking closer, I think there are only some keyword differences and the implementation is exactly the same.

So another suggestion: remove the `lookup` function, rename `eldoc` to lookup and provide some options to configure how the keys are named, depending on what op was used.

Reply:
I tried to merge the functionality into single lookup function, but I'm starting to have a feeling I'm walking in circles. The problems are (and have been from the very beginning) aliased namespaces in `(:require)`. 

"Code completion" doesn't work correctly:
```clj
(s/index-of )
;;         ^-- cursor here (after f)
;; Code completion (ctrl+space)
;; ns-str:  clojure.string
;; sym-str: s/index-of
;; ==> vars/current-ns becomes clojure.string, thus unable to resolve s/index via both ns-resolve and resolve
```
"Show Documentation" works OK:
```clj
(s/index-of )
;;        ^-- cursor here (before or after f)
;; Show documentation (ctrl+k ctrl+i)
;; ns-str:  core
;; sym-str: s/index-of
;; ==> vars/current-ns becomes core. Then (ns-resolve 'core 's/index-of) works fine
```
"Show parameters" works OK:
```clj
(s/index-of  )
;;          ^-- cursor here
;; Show parameters of function (ctrl+shift+space)
;;  
;; ns-str:  core
;; sym-str: s/index-of
;; ==> vars/current-ns becomes core. Then (ns-resolve 'core 's/index-of) works fine
```

TBH, at the moment I don't see any elegant way of supporting all these 3 scenarios (well, the "code completion" scenario is a trouble-maker) other than passing even ns-striped version of `sym-str` value and then trying to do yet one more "fallback ns-lookup attempt in case the first one that uses `ns-str` and `sym-str` wasn't able to resolve anything".

So basically something like:

```clj
(sci/binding [vars/current-ns (or sci-ns @vars/current-ns)]
        (let [m (sci/eval-string* ctx (format "
(let [sym-ns-str '%s
      sym-str '%s
      ns-striped-sym-str '%s]
  (when-let [v (or (ns-resolve sym-ns-str sym-str)
                   (ns-resolve sym-ns-str ns-striped-sym-str))]
    (let [m (meta v)]
      (assoc m :arglists (:arglists m)
       :doc (:doc m)
       :name (:name m)
       :ns (some-> m :ns ns-name)
       :val @v))))" ns-str sym-str sym-str-ns-striped))
```

This works, but I don't know if it's a good idea or not. I'm pushing it in a next fixup commit.